shell/platform/embedder/embedder.cc

```diff
@@ -1593,6 +1611,40 @@ static inline flutter::KeyEventType MapKeyEventType(
   return flutter::KeyEventType::kUp;
 }
 
+static FlutterEngineResult InternalSendPlatformMessage(
+    FLUTTER_API_SYMBOL(FlutterEngine) engine,
+    const char* channel,
+    const uint8_t* data,
+    size_t size,
+    FlutterDataCallback data_callback,
+    void* user_data) {
+  FlutterEngineResult result;
+
+  FlutterPlatformMessageResponseHandle* response_handle;
+  result = FlutterPlatformMessageCreateResponseHandle(
+      engine, data_callback, user_data, &response_handle);
+  if (result != kSuccess) {
+    return result;
+  }
+
+  const FlutterPlatformMessage message = {
+      sizeof(FlutterPlatformMessage),  // struct_size
+      channel,                         // channel
+      data,                            // message
+      size,                            // message_size
+      response_handle,                 // response_handle
+  };
+
+  result = FlutterEngineSendPlatformMessage(engine, &message);
+  FlutterEngineResult result2 =
+      FlutterPlatformMessageReleaseResponseHandle(engine, response_handle);
```

Comment:
nit: I think we can move this down below the if right? Also, please use a more descriptive name than `result2`, or, alternatively, just change the `return result2` to `return FlutterPlatofrmMessageReleaseResponseHandle(engine, response_handle);`.

Reply:
No. It's exactly what this awkward structure means: Even if the `SendPlatformMessage` call fails, we still need to release the response handle. I'll add a comment for it.

Reply:
Ahh ok, it's doing some kind of RAII thing here?

At any rate, it should be named better than `result2` :)